src/Keboola/DbWriter/Redshift/Application.php

```diff
@@ -90,20 +123,75 @@ public function loadFull($tableConfig, $manifest)
     private function getManifest($tableId)
     {
         return json_decode(file_get_contents(
-            $this['parameters']['data_dir'] . "/in/tables/" . $tableId . ".csv.manifest"
+            $this->container['parameters']['data_dir'] . "/in/tables/" . $tableId . ".csv.manifest"
         ), true);
     }
 
-    private function reorderColumnsFromManifest($manifestColumns, $items)
+    private function getInputMapping($tableId)
     {
-        $reordered = [];
-        foreach ($manifestColumns as $manifestCol) {
-            foreach ($items as $item) {
-                if ($manifestCol == $item['name']) {
-                    $reordered[] = $item;
-                }
+        foreach ($this->container['inputMapping'] as $inputTable) {
+            if ($tableId == $inputTable['source']) {
+                return $inputTable;
             }
         }
-        return $reordered;
+
+        throw new UserException(sprintf(
+            'Configuration mismatch. Table "%s" is missing from input mapping. Reloading the page and re-saving table configuration may fix the problem.',
+            $tableId
+        ));
+    }
+
+    /**
+     * Check if input mapping is really aligned with exported CSV file
+     *
+     * @param $inputMapping
+     * @param $manifest
+     * @throws UserException
+     */
+    private function checkColumns($inputMapping, $manifest)
```

Comment:
Jak jsme se bavili s columns v  manifestu bych vubec  nepracoval ani je nekontroloval. Pouzil bych jenom info z input mappingu a konfigurace, tam by naopak bylo dobre porovnat jestli sloupce sedi - k rozhozeni by mohlo dojit rucni manipulaci configu nebo treba nejakym bugem v ui.

Reply:
Ještě jsem radši kontroloval jestli aktuální stav nějak nepodporuje smazaný sloupec z tabulky (jako že by  ho to vynechalo. Ale není to tak, v ui extractoru se sice nezobrazuje ale když konfiguraci spustím tak docker runner vrátí user error o neexistujícím sloupci - takže to je issue spíš na UI. A tahle úprava by na tom neměla nic změnit.

Reply:
Nevím jestli nepočkat na fix v docker runneru https://github.com/keboola/docker-bundle/issues/62#issuecomment-361212870 ?